web/src/lib/utils/thumbnail-util.ts

```diff
@@ -43,33 +43,49 @@ export const getAltText = derived(t, ($t) => {
       return asset.exifInfo.description;
     }
 
-    let altText = $t('image_taken', { values: { isVideo: asset.type === AssetTypeEnum.Video } });
+    const date = fromLocalDateTime(asset.localDateTime).toLocaleString({ dateStyle: 'long' });
+    const hasPlace = !!asset.exifInfo?.city && !!asset.exifInfo?.country;
+    const names = asset.people?.filter((p) => p.name).map((p) => p.name) ?? [];
+    const peopleCount = names.length;
+    const isVideo = asset.type === AssetTypeEnum.Video;
 
-    if (asset.exifInfo?.city && asset.exifInfo?.country) {
-      const placeText = $t('image_alt_text_place', {
-        values: { city: asset.exifInfo.city, country: asset.exifInfo.country },
-      });
-      altText += ` ${placeText}`;
-    }
+    const values = {
+      date,
+      city: asset.exifInfo?.city,
+      country: asset.exifInfo?.country,
+      person1: names[0],
+      person2: names[1],
+      person3: names[2],
+      isVideo,
+      additionalCount: peopleCount > 3 ? peopleCount - 2 : 0,
+    };
 
-    const names = asset.people?.filter((p) => p.name).map((p) => p.name) ?? [];
-    if (names.length > 0) {
-      const namesText = $t('image_alt_text_people', {
-        values: {
-          count: names.length,
-          person1: names[0],
-          person2: names[1],
-          person3: names[2],
-          others: names.length > 3 ? names.length - 2 : 0,
-        },
-      });
-      altText += ` ${namesText}`;
+    if (peopleCount > 0) {
+      const key = getPlacePersonKey(peopleCount, hasPlace);
```

Comment:
This looks better. Similarly it is easier to find translations if they use $t('actual_key'). Can you just move the `$t` part to be inside the method, along with the keys? I don't think there is any benefit to separating them.

Reply:
Ok, this has been fixed. 

The `svelte-i18n` library doesn't export a type for the `$t` translation function, so there wasn't an easy way to pass a typed function closure into the method. Instead, I put the logic directly into the `getAltText` method.